packages/plugin-ext/src/main/browser/plugin-manager-client.ts

```diff
@@ -7,14 +7,46 @@
 
 import { injectable, inject } from 'inversify';
 import URI from '@theia/core/lib/common/uri';
-import { MessageService, Command } from '@theia/core/lib/common';
+import { MessageService, Command, Emitter, Event } from '@theia/core/lib/common';
 import { LabelProvider, isNative } from '@theia/core/lib/browser';
 import { WindowService } from '@theia/core/lib/browser/window/window-service';
 import { WorkspaceService } from '@theia/workspace/lib/browser';
 import { FileSystem } from '@theia/filesystem/lib/common';
 import { FileDialogFactory, DirNode } from '@theia/filesystem/lib/browser';
 import { HostedPluginServer } from '../../common/plugin-protocol';
 
+/**
+ * Commands to control Hosted plugin instances.
+ */
+export namespace HostedPluginCommands {
+    export const START: Command = {
+        id: 'hosted-plugin:start',
+        label: 'Hosted Plugin: Start Instance'
+    };
+    export const STOP: Command = {
+        id: 'hosted-plugin:stop',
+        label: 'Hosted Plugin: Stop Instance'
+    };
+    export const RESTART: Command = {
+        id: 'hosted-plugin:restart',
+        label: 'Hosted Plugin: Restart Instance'
+    };
+    export const SELECT_PATH: Command = {
+        id: 'hosted-plugin:select-path',
+        label: 'Hosted Plugin: Select Path'
+    };
+}
+
+/**
+ * Available states of hosted plugin instance.
+ */
+export type HostedPluginState =
+    'stopped'
+    | 'starting'
+    | 'running'
+    | 'stopping'
+    | 'failed';
+
```

Comment:
It will be better to use :
export enum HostedPluginState {
    Stopped = 'stopped',
    Starting = 'starting',
    Running = 'running',
    Stopping= 'stopping',
    Failed = 'failed'
}

Reply:
Thanks!